apps/arweave/src/ar_mining_worker.erl

```diff
@@ -112,31 +137,40 @@ init({Partition, PackingDifficulty}) ->
 	Name = name(Partition, PackingDifficulty),
 	?LOG_DEBUG([{event, mining_debug_worker_started},
 		{worker, Name}, {pid, self()}, {partition, Partition}]),
-	gen_server:cast(self(), handle_task),
-	gen_server:cast(self(), check_worker_status),
-	prometheus_gauge:set(mining_server_chunk_cache_size, [Partition], 0),
-	{ok, #state{ name = Name, partition_number = Partition,
-			is_pool_client = ar_pool:is_client(), packing_difficulty = PackingDifficulty }}.
+	ChuckCache = ar_mining_cache:new(),
+	State0 = #state{
+		name = Name,
+		chunk_cache = ChuckCache,
+		partition_number = Partition,
+		is_pool_client = ar_pool:is_client(),
+		packing_difficulty = PackingDifficulty
+	},
+	gen_server:cast(self(), ?MSG_HANDLE_TASK),
+	gen_server:cast(self(), ?MSG_CHECK_WORKER_STATUS),
+	{ok, report_chunk_cache_metrics(State0)}.
 
 handle_call(Request, _From, State) ->
 	?LOG_WARNING([{event, unhandled_call}, {module, ?MODULE}, {request, Request}]),
 	{reply, ok, State}.
 
-handle_cast({set_difficulty, DiffPair}, State) ->
-	{noreply, State#state{ diff_pair = DiffPair }};
+handle_cast(?MSG_SET_CACHE_LIMITS(SubchunkCacheLimitBytes, VDFQueueLimit), State) ->
```

Comment:
```suggestion
handle_cast(?MSG_SET_CACHE_LIMITS(ChunkCacheLimitBytes, VDFQueueLimit), State) ->
```

Reply:
`CacheLimitBytes` would be better, probably?

Reply:
CacheLimitBytes sounds good.